pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1953,3 +1968,72 @@ def _check_unnecessary_dict_index_lookup(
                             node=subscript,
                             args=("1".join(value.as_string().rsplit("0", maxsplit=1)),),
                         )
+
+    def _check_unnecessary_list_index_lookup(
+        self, node: Union[nodes.For, nodes.Comprehension]
+    ) -> None:
+        if (
+            not isinstance(node.iter, nodes.Call)
+            or not isinstance(node.iter.func, nodes.Name)
+            or not node.iter.func.name == "enumerate"
+            or not isinstance(node.iter.args[0], nodes.Name)
+        ):
+            return
+
+        if not isinstance(node.target, nodes.Tuple) or len(node.target.elts) < 2:
+            # enumerate() result is being assigned without destructuring
+            return
+
+        iterating_object_name = node.iter.args[0].name
```

Comment:
I'm wondering if there are edge cases to consider with `len(args) > 1`, but I think in those cases using the index is so unnecessarily complicated that most people will have moved to using `value` at that point. So I think we can leave those edge cases unconsidered here.